pkg/skaffold/jib/jib.go

```diff
@@ -17,14 +17,86 @@ limitations under the License.
 package jib
 
 import (
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v1alpha3"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
+
 	"github.com/pkg/errors"
 )
 
-func GetDependenciesMaven(_ /*workspace*/ string, _ /*a*/ *v1alpha3.JibMavenArtifact) ([]string, error) {
-	return nil, errors.New("jib maven support is unimplemented")
+// GetDependenciesMaven finds the source dependencies for the given jib-maven artifact.
+// All paths are absolute.
+// TODO(coollog): Add support for multi-module projects.
+func GetDependenciesMaven(workspace string, a *v1alpha3.JibMavenArtifact) ([]string, error) {
+	executable, subCommand := getCommandMaven(workspace, a)
+	return getDependencies(workspace, "pom.xml", executable, subCommand, "jib-maven")
+}
+
+// GetDependenciesGradle finds the source dependencies for the given jib-gradle artifact.
+// All paths are absolute.
+func GetDependenciesGradle(workspace string, a *v1alpha3.JibGradleArtifact) ([]string, error) {
+	executable, subCommand := getCommandGradle(workspace, a)
+	return getDependencies(workspace, "build.gradle", executable, subCommand, "jib-gradle")
+}
+
+// resolveFile resolves the absolute path of the file named filename in directory workspace, erroring if it is not a file
+func resolveFile(workspace string, filename string) (string, error) {
```

Comment:
- this could live in `pkg/skaffold/util/util.go` 
- I would rename it to just `AbsFile`. 
- `workspace` -> `dir`